### PR TITLE
Add simple example Jupyter notebook (for Binder)

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Towards #126

Preview Link:
https://mybinder.org/v2/gh/EricCousineau-TRI/pygccxml/feature-py-notebook-example?filepath=pygccxml/docs/examples/notebook/example.ipynb

### CODE REVIEW

**Comment by tao558** [comment]
Just as a note: this looks good so far. I'll do another review and approve when the `todos` are gone

**Comment by EricCousineau-TRI** [comment]
Removed TODO about path stuff (I won't be able to investigate anytime soon).

Briefly trying the JupyterLab setup, but may descope that as well.

**Comment by EricCousineau-TRI** [comment]
JupyterLab does not work easily; will leave in TODO; breadcrumb commit: 608f224

I think the remaining TODOs can remain as they are.

@tao558 Might you have time to take a look?

**Comment by tao558** [comment]
@EricCousineau-TRI I'll take a look ASAP

File: .dockerignore
Comment (tao558) [inline .dockerignore:1]: Is this file required in addition to the `.dockerignore` in the `.binder` directory?

File: .binder/README.md
Comment (tao558) [inline .binder/README.md:14]:
```suggestion
following `build` and `run` commands from the top level of this Git
```

File: .binder/README.md
Comment (tao558) [inline .binder/README.md:18]: Is this needed? Assuming we're already at the top level of the git repo, this would put is at `pygccxml/pygccxml`. Then `.binder` would be in the parent directory. Excluding this `cd` call works for me

File: .binder/README.md
Comment (tao558) [inline .binder/README.md:24]: Do all 3 of the displayed links work for you? Only the ` http://127.0.0.1:8888...` link works for me.

**Review by tao558 — CHANGES_REQUESTED** [review]

File: .dockerignore
Comment (EricCousineau-TRI) [reply .dockerignore:1]: Sorry for the delay! But yes, this is important to ignore similar artifacts for this directory. `.binder/.dockerignore` functions the same as `.gitignore` - it only applies to enclosed directories.

File: .binder/README.md
Comment (EricCousineau-TRI) [reply .binder/README.md]: Done.

File: .binder/README.md
Comment (EricCousineau-TRI) [reply .binder/README.md]: Done. Removed.

File: .binder/README.md
Comment (EricCousineau-TRI) [reply .binder/README.md]: Oops! Good point, updated text.

File: setup.py
Comment (EricCousineau-TRI) [reply setup.py]: Done. Changed to `notebook`.

**Comment by EricCousineau-TRI** [comment]
FYI I changed requirements away from use a `set` literal which is then converted to a `list`. That operation is redundant when `dict/set` become order-preserving (Python 3.5, I believe?), and provides information loss in lower Python versions.

File: .binder/README.md
Comment (tao558) [inline .binder/README.md:24]: Any way to either make the rest of the urls work, or exclude the broken urls from the printed list?

**Review by tao558 — COMMENTED** [review]
Other than the one comment looks good. Are you OK with me squashing the commits when merging?

File: .binder/README.md
Comment (EricCousineau-TRI) [reply .binder/README.md:24]: I don't think so / don't think it's worth spending time on.

This is just for local development; actually using binder.org will shield users from it.

**Comment by EricCousineau-TRI** [comment]
> Are you OK with me squashing the commits when merging?

Yup!

File: .binder/README.md
Comment (tao558) [inline .binder/README.md:31]: Does this need to be in double quotes (")? Additionally, I get the error response `docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"-v\": executable file not found in $PATH": unknown.`

Comment (EricCousineau-TRI) [reply .binder/README.md]: Nah, just needs to be reordered. If it's still confusing, it'd be best just to remove it, and leave it up to others to inspect `docker run` help.

 See usage string: `docker run [options...] IMAGE [COMMAND] [ARG...]`
It's because `-v` is for `docker`, but if you put it at the end, it's interpreted as `COMMAND` and overrides the entrypoint. Refer to docker docs for more info.

Comment (tao558) [reply .binder/README.md]: Could you just clarify that this is for `docker-run`? Then we're good to merge!

Comment (EricCousineau-TRI) [reply .binder/README.md]: Done! edfa8d8

**Comment by EricCousineau-TRI** [comment]
Please squash-merge if that's an option. If not, I can squash myself since review is done.

**Review by tao558 — APPROVED** [review]

**Comment by tao558** [comment]
Thanks @EricCousineau-TRI! Good to finally get this merged.

**Pull request merged**
